src/http-middlewares/before/add-query-params.js

```diff
@@ -2,13 +2,34 @@
 
 const querystring = require('querystring');
 
+// Mutates the object (it'll be deleted later anyway)
+const removeEmptyParams = params =>
+  Object.keys(params).map(param => {
+    if (
+      params[param] === '' ||
+      params[param] === null ||
+      typeof params[param] === 'undefined'
+    ) {
+      delete params[param];
+    }
+  });
+
 // Take params off of req.params and append to url - "?a=1&b=2"".
 // This middleware should run *after* custom middlewares, because
 // custom middlewares might add params.
 const addQueryParams = req => {
   if (Object.keys(req.params || {}).length) {
     const splitter = req.url.indexOf('?') === -1 ? '?' : '&';
-    req.url += splitter + querystring.stringify(req.params);
+
+    if (req.omitEmptyParams === true) {
+      removeEmptyParams(req.params);
+    }
+
+    const stringifiedParams = querystring.stringify(req.params);
+
+    if (stringifiedParams) {
+      req.url += `${splitter}${stringifiedParams}`;
```

Comment:
I think this is fine as is, but I think we’d be better off parsing the url with the stdlib, pulling out the query, merging with new params, and adding the whole thing rather than guessing the splitter and adding it that way. 

Again, it’s _fine_, but using established tools feels a little more above the board.

Reply:
Agreed. I just didn't want to change too much here, honestly.

We only test the `splitter` now by not adding `?` (it was in the case the params existed but where then removed).